### PR TITLE
Graph Dijkstra

### PR DESCRIPTION
**Pull request by Yag000**
Add python implementation of Dijkstra's algorithm

### CODE REVIEW

File: content/graph/GraphClass.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+class Graph:
+    def __init__(self, v, egdes=None):
+        self.v = v
+        self.edges = defaultdict(dict)
+        if egdes:
+            for u, v, w in egdes:
+                self.add_edge(u, v, w)
+
+    def add_edge(self, u, v, w):
```
Comment (GabinDDL) [inline content/graph/GraphClass.py:16]: do we assume we'll only have non-oriented graphs?

**Comment by GabinDDL** [comment]
apart from the comment I left, the mr is good to me

File: content/graph/GraphClass.py (hunk 1 above)
Comment (Yag000) [reply content/graph/GraphClass.py:16]: It's actually a good point, I should change that I did this in order to solve an problem on a non oriented graph and I left it there) I'll change that later (tomorrow I think)

Comment (Yag000) [reply content/graph/GraphClass.py:16]: Should be fixed now :)

**Review by GabinDDL — APPROVED** [review]
it's good for me

**Review by YanisLcn — COMMENTED** [review]
LGTM. Nice job. 
Nevertheless, shouldn't we also add the predecessors in the dijkstra's algorithm to keep track of the path to follow ?

**Comment by Yag000** [comment]
> LGTM. Nice job. Nevertheless, shouldn't we also add the predecessors in the dijkstra's algorithm to keep track of the path to follow ?

That could be a nice addition, but I do not have time to implement it right now. I'll merge this (since it contains the basics for working with graphs) and we could do an pr later for that.

**Pull request merged**